+bids/layout.m

```diff
@@ -456,66 +453,120 @@ function tolerant_message(use_schema, msg)
+      BIDS.subjects(info_src.sub_idx).(info_src.modality)(info_src.file_idx).intended_for = ...
+          intended_for;
+
+      % update "dependency" field of target file
+      informed_by = [];
+      for iTargetFile = 1:size(intended_for, 1)
+        info_tgt = bids.internal.return_file_info(BIDS, intended_for{iTargetFile});
+
+        % TODO: this should probably check that the informed_by field does not
+        % already exist in the structure
+        % TODO: This assumes that a file will only be informed by a single file from
+        % this moodality
+        informed_by.(info_src.modality) = file_list{iFile};
+        BIDS.subjects(info_tgt.sub_idx).(info_tgt.modality)(info_tgt.file_idx).informed_by = ...
+            informed_by;
+      end
 
-    p = bids.internal.file_utils('FPList', pth,  ['^' strrep(filename, '.tsv', '\.json') '$']);
-    if ~isempty(p)
-      structure.meta = bids.util.jsondecode(p);
     end
 
   end
 
 end
 
-function structure = manage_intended_for(structure, subject, pth)
+function [metadata, intended_for] = check_intended_for_exist(subject, filename)
 
-  if isempty(structure.meta)
+  metadata =  bids.internal.get_metadata(filename);
+
+  intended_for = {};
+
+  if isempty(metadata) || ~isfield(metadata, 'IntendedFor')
+    warning('Missing field IntendedFor for %s', filename);
     return
 
   else
 
-    % Get all NIfTIs that this m0scan is intended for
     path_intended_for = {};
-    if ~isfield(structure.meta, 'IntendedFor')
-      warning('Missing field IntendedFor for %s', structure.filename);
 
-    elseif ischar(structure.meta.IntendedFor)
-      path_intended_for{1} = structure.meta.IntendedFor;
+    if ischar(metadata.IntendedFor)
+      path_intended_for{1} = metadata.IntendedFor;
 
-    elseif isstruct(structure.meta.IntendedFor)
-      for iPath = 1:length(structure.meta.IntendedFor)
-        path_intended_for{iPath} = structure.meta.IntendedFor(iPath); %#ok<*AGROW>
+    elseif isstruct(metadata.IntendedFor)
+      for iPath = 1:length(metadata.IntendedFor)
+        path_intended_for{iPath} = metadata.IntendedFor(iPath); %#ok<*AGROW>
       end
 
     end
+  end
 
-    for iPath = 1:length(path_intended_for)
-      % check if this NIfTI is not missing
-      if ~exist(fullfile(fileparts(pth), path_intended_for{iPath}), 'file')
-        warning(['Missing: ' fullfile(fileparts(pth), path_intended_for{iPath})]);
+  for iPath = 1:length(path_intended_for)
 
-      else
-        % also check that this NIfTI aims to the same m0scan
-        [~, path2check, ext2check] = fileparts(path_intended_for{iPath});
-        filename_found = max(arrayfun(@(x) strcmp(x.filename, ...
-                                                  [path2check ext2check]), ...
-                                      subject.perf));
-        if ~filename_found
-          warning(['Did not find NIfTI for which is intended: ' structure.filename]);
+    file_path = strrep(subject.path, subject.session, '');
```

Comment:
Unsafe: If `subject.path` contains two or more occurrences of `subject.session`, they will be both replaced, making path unusable. Only last occurrence must be replaced.

Also in bids intended for paths are stored in unix way, with `/` separator. The code may be run on windows system with `/` separators. I'm not sure that it will not cause problems during path generation.

Reply:
> Also in bids intended for paths are stored in unix way, with `/` separator. The code may be run on windows system with `/` separators. I'm not sure that it will not cause problems during path generation.

Good catch. Will update that.

Reply:
> Unsafe: If `subject.path` contains two or more occurrences of `subject.session`, they will be both replaced, making path unusable. Only last occurrence must be replaced.

Ha yes! Thanks!